src/Evolution/Systems/Cce/OptionTags.hpp

```diff
@@ -204,8 +209,13 @@ struct ExtractionRadius : db::SimpleTag {
   using option_tags = tmpl::list<OptionTags::ExtractionRadius>;
 
   static constexpr bool pass_metavariables = false;
-  static double create_from_options(const double extraction_radius) noexcept {
-    return extraction_radius;
+  static double create_from_options(
+      const std::optional<double> extraction_radius) noexcept {
+    ASSERT(
+        static_cast<bool>(extraction_radius),
+        "Cannot infer an `\"Auto\"` value of ExtractionRadius in this context "
+        "-- please specify an explicit value in the input file.");
```

Comment:
I think you can avoid this situation by switching the `OptionTags::ExtractionRadius` at compile-time. If I understand correctly, you will either create the `Tags::ExtractionRadius` or the `Tags::H5WorldtubeBoundaryDataManager` from options, never both (since if you created both from options you could never set the extraction radius to "Auto"). So you can add a boolean template parameter to `OptionTags::ExtractionRadius` that has the effect of toggling its type between `double` and `Options::Auto<double>` (and changes the help string), or just create a separate `OptionTags::ExtractionRadiusAuto`. Then you can use the non-auto version here in `Tags::ExtractionRadius` and the auto-version in `Tags::H5WorldtubeBoundaryDataManager`.